src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,173 @@
     }
 
     /**
      * Given a String representation of a date and some timezone,
      * return LocalDateTime in this timezone.
      */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateInString)
+                .withZoneSameInstant(ZoneId.of(zone));
+        return zonedDateTime.toLocalDateTime();
     }
 
     /**
      * Given some LocalDateTime, return an OffsetDateTime with the local time offset applied
      * (`+02:00` for Ukraine).
-     *
+     * <p>
      * Example: we receive a LocalDateTime with a value `2019-09-06T13:17`.
-     *          We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
-     *          where `+02:00` is the offset for our local timezone.
-     *
+     * We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
+     * where `+02:00` is the offset for our local timezone.
+     * <p>
      * OffsetDateTime is recommended to use for storing date values in a database.
      */
-    public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+    public OffsetDateTime offsetDateTime(LocalDateTime localDateTime) {
+        final ZoneId zone = ZoneId.of("Europe/Paris");
+        ZoneOffset zoneOffSet = zone.getRules().getOffset(localDateTime);
+        return localDateTime.atOffset(zoneOffSet);
     }
 
     /**
      * Given a String formatted as `yyyymmdd`,
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd");
+            return Optional.ofNullable(LocalDate.parse(date, formatter));
+        } catch (DateTimeException e) {
+            return Optional.empty();
+        }
     }
 
     /**
      * Given a String formatted as `d MMM yyyy` (MMM - Sep, Oct, etc),
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> customParseDate(String date) {
-        return Optional.empty();
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d MMM yyyy", Locale.ENGLISH);
+            return Optional.of(LocalDate.parse(date, formatter));
+        } catch (DateTimeException e) {
+            return Optional.empty();
+        }
     }
 
     /**
      * Given some LocalDateTime, return a String formatted as
      * `day(2-digit) month(full name in English) year(4-digit) hours(24-hour format):minutes`.
-     *
+     * <p>
      * Example: "01 January 2000 18:00".
      */
     public String formatDate(LocalDateTime dateTime) {
-        return "";
+        DateTimeFormatter europeanDateFormatter = DateTimeFormatter
+                .ofPattern("dd MMMM yyyy HH:mm", Locale.ENGLISH);
+        return europeanDateFormatter.format(dateTime);
```

Comment:
```suggestion
        return dateTime.format(DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm"));
```
but add `"dd MMMM yyyy HH:mm"` to constant

Reply:
fixed